pybind_interface/avx2/CMakeLists.txt

```diff
@@ -4,7 +4,7 @@ project(qsim)
 IF (WIN32)
     set(CMAKE_CXX_FLAGS "/arch:AVX2 /O2 /openmp")
 ELSE()
-    set(CMAKE_CXX_FLAGS "-mavx2 -mfma -O3 -fopenmp")
+    set(CMAKE_CXX_FLAGS "-mavx2 -mfma -O3")
```

Comment:
It seems this breaks OpenMP support on Linux.

Reply:
How can you tell that?

AFAIK the `find(OpenMP REQUIRED)` will add the `-fopenmp` flag.

Reply:
I run `pip install .`. The `-fopenmp` flag is not added  and the code is compiled without OpenMP. According to [An Introduction to Modern CMake](https://cliutils.gitlab.io/modern-cmake/chapters/packages/OpenMP.html), an additional line is needed to add the `-fopenmp` flag.

Reply:
Thanks @sergeisakov, I had just confirmed this indeed after running the tests.

I have now fixed it and tested it locally and ensured that OpenMP is linked.